pages/treeherder.py

```diff
@@ -51,6 +59,42 @@ def open_perfherder_page(self):
         from perfherder import PerfherderPage
         return PerfherderPage(self.base_url, self.selenium).wait_for_page_to_load()
 
+    def select_next_job(self):
+        el = self.selenium.find_element(*self._resultset_locator)
+        Wait(self.selenium, self.timeout).until(EC.visibility_of(el))
+        el.send_keys(Keys.ARROW_RIGHT)
+        Wait(self.selenium, self.timeout).until(lambda s: self.job_result_status)
+        return self.selenium.find_element(*self._next_job_title).get_attribute('title')
+
+    def add_selected_job_to_pinboard(self):
+        el = self.selenium.find_element(*self._resultset_locator)
+        Wait(self.selenium, self.timeout).until(EC.visibility_of(el))
+        el.send_keys(Keys.SPACE)
+        Wait(self.selenium, self.timeout).until(lambda s: self.is_pinboard_open)
+
+    @property
+    def is_pinboard_open(self):
+        return self.is_element_visible(self._pinboard_locator)
+
+    @property
+    def pinned_job_title(self):
+        return self.selenium.find_element(*self._next_job_title).get_attribute('title')
+
+    def pin_job_from_logviewer(self):
```

Comment:
We've already used 'logviewer' to refer to the new window that displayed the job's logs. I would call this `job_details_panel` or similar.